### PR TITLE
feat(gradient-aws-kubernetes): refactor

### PR DESCRIPTION
**Pull request by marquiswashere**
- Light Variable/Locals Refactoring
- https://github.com/Paperspace/gradient-installer/issues/243 is addressed
- README init for module (I would like to begin to bubble up module inputs and outputs)
- Adding `worker_group` `list(map(any))` as module output

### CODE REVIEW

File: gradient-aws/modules/kubernetes/cpu-instance-vars.tf [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    default = {
+        aws_ec2_instance_type : "c5.xlarge",
+        paperspace_type : "cpu",
+        instance_type_metadata : {
+        # Services Small
+        "services-small" : {
+            instance_type_name : "services-small"
+
+            default_node_asg_capacities : {
+            "desired" : 0
```
Comment (bbatha) [inline gradient-aws/modules/kubernetes/cpu-instance-vars.tf:17]: Can you run terraform-fmt over this, the vscode terraform plugin should include it.

File: gradient-aws/modules/kubernetes/cpu-instance-vars.tf [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            "max" : 600
+            }
+            node_pool_type : "cpu"
+            root_storage_volume_type : "gp2"
+
+            root_volume_size : 50
+            kubelet_extra_args : []
+            },
+        # Tensorboard CPU Small
+        "tensorboard-cpu-small" : {
```
Comment (bbatha) [inline gradient-aws/modules/kubernetes/cpu-instance-vars.tf:73]: We shouldn't need these at all `-type-size` labels at all, unless we need some kind of backwards compatibility map. Even then we're not even 1.0 so I'm ok with breaking compat so long as we write up a migration doc.

The instance-type names should always be the name of the aws instance type name, e.g. p2.xlarge, except for `services-small` which should map to the new size

File: gradient-aws/modules/kubernetes/cpu-instance-vars.tf [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            kubelet_extra_args : []
+            },
+        # Notebook CPU Small
+        "notebook-cpu-small" : {
+            instance_type_name : "notebook-cpu-small"
+
+            default_node_asg_capacities : {
+            "desired" : 0
+            "min" : 1
+            "max" : 600
```
Comment (bbatha) [inline gradient-aws/modules/kubernetes/cpu-instance-vars.tf:64]: These seem high for defaults, maybe lower to start with

Comment (paperspacemarquis) [reply gradient-aws/modules/kubernetes/cpu-instance-vars.tf:64]: @bbatha For sane defaults what are you thinking for the max I'd say arbitrarily maybe like 20?
The 600 is fro (https://github.com/Paperspace/gradient-installer/blob/master/gradient-aws/modules/kubernetes/main.tf#L14)

File: gradient-aws/modules/kubernetes/cpu-instance-vars.tf (hunk 2 above)
Comment (bbatha) [reply gradient-aws/modules/kubernetes/cpu-instance-vars.tf:73]: I wouldn't worry about creating a bunch of issues AFAIK there are only 3 aws clusters, the staging one, a poc we just wrapped up, and a customer on a really old one that's going to need more manual upgrades as well (there are some kube version bumps). But you're welcome to keep it if it makes the change easier.

File: gradient-aws/modules/kubernetes/cpu-instance-vars.tf (hunk 3 above)
Comment (bbatha) [reply gradient-aws/modules/kubernetes/cpu-instance-vars.tf:64]: I guess keep it if it was already there

File: gradient-aws/modules/kubernetes/cpu-instance-vars.tf (hunk 2 above)
Comment (bbatha) [reply gradient-aws/modules/kubernetes/cpu-instance-vars.tf]: Sounds like a plan

File: gradient-aws/modules/kubernetes/cpu-instance-vars.tf [hunk 4]
```diff
@@ -0,0 +1,158 @@
+# CPU EC2 Instance Types
+# Within this variables file each variable points to a specific EC2 Instace Type
```
Comment (bk-paperspace) [inline gradient-aws/modules/kubernetes/cpu-instance-vars.tf:2]: typo: s/Instace/Instance

File: gradient-aws/modules/kubernetes/gpu-instance-vars.tf [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+          "min" : 0
+          "max" : 20
+        }
+        node_pool_type : "gpu"
+        root_storage_volume_type : "gp2"
+
+        root_volume_size : 50
+        kubelet_extra_args : [
+          "cloud.google.com/gke-accelerator=nvidia-tesla-v100",
+        "k8s.amazonaws.com/accelerator=nvidia-tesla-v100"]
```
Comment (bk-paperspace) [inline gradient-aws/modules/kubernetes/gpu-instance-vars.tf:27]: Was `terraform fmt` run?

File: gradient-aws/modules/kubernetes/gpu-instance-vars.tf [hunk 6]
```diff
@@ -0,0 +1,274 @@
+# GPU EC2 Instance Types
+# Within this variables file each variable points to a specific EC2 Instace Type
```
Comment (bk-paperspace) [inline gradient-aws/modules/kubernetes/gpu-instance-vars.tf:2]: typo: s/Instace/Instance

File: modules/gradient-processing/templates/values.yaml.tpl [hunk 7]
```diff
@@ -487,6 +487,27 @@ argo:
     nodeSelector:
       paperspace.com/pool-name: ${service_pool_name}
 
+argo-rollouts:
+  controller:
+    nodeSelector:
+      paperspace.com/pool-name: ${service_pool_name}
+%{ if is_public_cluster }
+    resources:
```
Comment (bk-paperspace) [inline modules/gradient-processing/templates/values.yaml.tpl:495]: is this `resources` field not needed on the else case (between L502/3) below?

Comment (marquiswashere) [reply modules/gradient-processing/templates/values.yaml.tpl:495]: I think it's okay, this is what's in master (https://github.com/Paperspace/gradient-installer/blob/master/modules/gradient-processing/templates/values.yaml.tpl#L490-L509), I didn't rebase my branch well.

**Review by bk-paperspace — APPROVED** [review]

**Pull request merged**